### DISCUSSION TITLE
How to exclude directory if not empty?

### DISCUSSION THREAD

**Discussion by EloB** [sharkdp/fd#1509] [Q&A]
I'm trying to zip and passing both directory and files makes the archive bad.

```bash
fd -p -g "$PWD/packages/*/dist/**/*" -X tar --zstd -cf my.tar.zst {}
```

The problem by doing so is that tar will append the files twice because I pass both file and directory.

**Current behaviour**
```
packages/hello/dist/empty-folder/
packages/hello/dist/something/
packages/hello/dist/something/a.txt
packages/hello/dist/something/b.txt
```

**Wanted behaviour**
```
packages/hello/dist/empty-folder/
packages/hello/dist/something/a.txt
packages/hello/dist/something/b.txt
```

**Comment by tmccombs** [comment]
you can use `-t f` to limit the output of fd to files.

**Reply by EloB** [reply]
Thanks for your reply. The problem is that I might want empty directories too. :)

**Comment by tavianator** [comment]
I don't think there's a way to get files + empty directories with one `fd` invocation.  But for your concrete problem, you could try

```console
$ fd -p -g "$PWD/packages/*/dist" -X tar --zstd -cf my.tar.zst {}
```

or

```console
$ fd -p -g "$PWD/packages/*/dist/**/*" -X tar --no-recursion --zstd -cf my.tar.zst {}
```
